fulu/gp_aug.py

```diff
@@ -49,14 +56,19 @@ def fit(self, t, flux, flux_err, passband):
         log_lam  = add_log_lam(passband, self.passband2lam)
 
         X = np.concatenate((t.reshape(-1, 1), log_lam.reshape(-1, 1)), axis=1)
-
+        
+        X_error = (flux_err / np.std(flux)).reshape(-1)
         self.ss = StandardScaler()
         X_ss = self.ss.fit_transform(X)
 
-        kernel = C(1.0) * RBF([1.0, 1.0]) + WhiteKernel()
-        self.reg = GaussianProcessRegressor(kernel=kernel, normalize_y=True, n_restarts_optimizer=5,
-                                            optimizer="fmin_l_bfgs_b", random_state=42)
+        self.kernel = C(1.0) * RBF([1.0, 1.0]) + WhiteKernel()
 
+        if flag_err:
+            self.reg = GaussianProcessRegressor(kernel=kernel, alpha=X_error, optimizer="fmin_l_bfgs_b", n_restarts_optimizer=5, normalize_y=True, random_state=42)
+
+        else:
+            self.reg = GaussianProcessRegressor(kernel=kernel, optimizer="fmin_l_bfgs_b", n_restarts_optimizer=5, normalize_y=True, random_state=42)
+            
```

Comment:
Please consider simplification to make it more maintainable, for example:

```python
reg_kwargs = dict(kernel=kernel, optimizer="fmin_l_bfgs_b", n_restarts_optimizer=5, normalize_y=True, random_state=42)
if flag_err:
    reg_wargs[alpha] = X_error
self.reg = GaussianProcessRegressor(**reg_kwargs)
```

Reply:
Updated due to this recommendation.